### PR TITLE
Adding unit tests

### PR DESCRIPTION
**Pull request by lucifercr07**
Unit tests being added for the following files:
- mutation-tracker.js
- property-list.js
- request.js
- url.js
- variable.js

### CODE REVIEW

File: test/unit/variable.test.js [hunk 1]
```diff
@@ -166,6 +166,18 @@ describe('Variable', function () {
         expect(v1.value).to.equal('{"foo":"bar"}');
     });
 
+    it('should set value to null when value provided as cyclic object value (json)', function () {
+        var obj = {};
+        obj.prop = obj;
+        // eslint-disable-next-line one-var
```
Comment (codenirvana) [inline test/unit/variable.test.js:172]: why?

File: test/unit/variable.test.js [hunk 2]
```diff
@@ -166,6 +166,18 @@ describe('Variable', function () {
         expect(v1.value).to.equal('{"foo":"bar"}');
     });
 
+    it('should set value to null when value provided as cyclic object value (json)', function () {
```
Comment (codenirvana) [inline test/unit/variable.test.js:169]:
```suggestion
    it('should set cyclic object value to null when type is provided (json)', function () {
```

File: test/unit/url.test.js [hunk 3]
```diff
@@ -57,6 +57,14 @@ describe('Url', function () {
                 expect(url.update).to.be.ok;
                 expect(url.update).to.be.a('function');
             });
+
+            it('should handle query parameters as string', function () {
```
Comment (codenirvana) [inline test/unit/url.test.js:61]: why is this in the `has property` block?

File: test/unit/url.test.js [hunk 4]
```diff
@@ -1353,6 +1361,18 @@ describe('Url', function () {
             expect(url.getQueryString()).to.equal('');
         });
 
+        it('should return the stringified query string for given URL', function () {
```
Comment (codenirvana) [inline test/unit/url.test.js:1364]: `given URL` or when the query parameters are set.

File: test/unit/request.test.js [hunk 5]
```diff
@@ -581,6 +592,56 @@ describe('Request', function () {
...
+                            key: 'some',
+                            value: 'header'
+                        },
+                        {
+                            key: 'other',
+                            value: 'otherheader',
+                            disabled: true
+                        }
+                    ],
+                    auth: null
```
Comment (codenirvana) [inline test/unit/request.test.js:611]: set a valid auth

File: test/unit/request.test.js [hunk 6]
```diff
@@ -581,6 +592,56 @@ describe('Request', function () {
...
+                            value: 'header'
+                        },
+                        {
+                            key: 'other',
+                            value: 'otherheader',
+                            disabled: true
+                        }
+                    ],
+                    auth: {
+                        type: 'type'
```
Comment (codenirvana) [inline test/unit/request.test.js:636]: set a valid auth first.

File: test/unit/request.test.js [hunk 7]
```diff
@@ -687,4 +748,39 @@ describe('Request', function () {
             expect(request.toJSON()).to.have.property('method', 'POSTMAN');
         });
     });
+
+    describe('addHeader', function () {
+        it('should add a header to the PropertyList of headers', function () {
```
Comment (codenirvana) [inline test/unit/request.test.js:753]:
```suggestion
        it('should add a header to the request', function () {
```

File: test/unit/property-list.test.js [hunk 8]
```diff
@@ -960,6 +963,24 @@ describe('PropertyList', function () {
                 val: 'value2'
             }]);
         });
+
+        it('should throw error when unable to upsert into a list Type that does not support .update()', function () {
+            FakeType.prototype.update = null;
```
Comment (codenirvana) [inline test/unit/property-list.test.js:968]: Don't rely on the global variable.

File: test/unit/property-list.test.js [hunk 9]
```diff
@@ -1033,6 +1054,25 @@ describe('PropertyList', function () {
                 val: 'value2'
             }]);
         });
+
+        it('should return in case of list members not array', function () {
+            // @todo Not working as expected not covering line 340
```
Comment (codenirvana) [inline test/unit/property-list.test.js:1059]: why? 🤔

File: test/unit/property-list.test.js [hunk 10]
```diff
@@ -1201,4 +1241,24 @@ describe('PropertyList', function () {
...
     });
+
+    describe('.toString', function () {
+        var FakeType = function (opts) {
+            _.assign(this, opts);
+        };
+        FakeType._postman_propertyIndexKey = 'key';
+        FakeType.prototype.update = function (opts) {
+            _.assign(this, opts);
+        };
```
Comment (codenirvana) [inline test/unit/property-list.test.js:1252]: Move this to dependent `it` block.

File: test/unit/property-list.test.js [hunk 11]
```diff
@@ -1201,4 +1241,24 @@ describe('PropertyList', function () {
...
+
+        it('should handle when unparse method not defined in Type', function () {
```
Comment (codenirvana) [inline test/unit/property-list.test.js:1254]: add another test where it uses prototype's `toString`.

**Review by codenirvana — CHANGES_REQUESTED** [review]

File: test/unit/variable.test.js (hunk 1 above)
Comment (lucifercr07) [reply test/unit/variable.test.js:172]: Can't define two variables separately, need to initialize `obj` first before using.

File: test/unit/request.test.js (hunk 5 above)
Comment (lucifercr07) [reply test/unit/request.test.js:611]: The test is for auth request when the type is null

File: test/unit/property-list.test.js (hunk 8 above)
Comment (lucifercr07) [reply test/unit/property-list.test.js:968]: Created a local variable for `FakeType`.

File: test/unit/property-list.test.js [hunk 12]
```diff
@@ -911,6 +911,9 @@ describe('PropertyList', function () {
         FakeType.prototype.update = function (opts) {
             _.assign(this, opts);
         };
+        FakeType.prototype.create = function (opts) {
```
Comment (codenirvana) [inline test/unit/property-list.test.js:914]: ?

Comment (lucifercr07) [reply test/unit/property-list.test.js:914]: Fixed.

File: test/unit/property-list.test.js (hunk 9 above)
Comment (lucifercr07) [reply test/unit/property-list.test.js:1059]: Working now as expected

**Comment by codenirvana** [comment]
No point in reviewing this PR after force-push, please create a new PR.

**Pull request closed without merging**